pkg/controller/kabaneroplatform/deployments.go

```diff
@@ -0,0 +1,139 @@
+package kabaneroplatform
```

Comment:
This looks ok, but here are some suggestions to handle now or in a future PR:
1. Since i foresee us adding many more resource types, it would be better if we put this file into it's own dir/package (i.e. kabaneroplatform/resources).
2. getOwnerReference() should be either public (or not in same package) or it should be in it's own Utils type part because i also see being used by other types.
3. In my opinion, we should have a template deployment and setters that callers can use. The setters would handle default/invalid values.